feat: add docker compose with docs

paolofacchinetti opened the pull request:
## Changes
- porting docker-compose.yml from the [local-reth](https://github.com/paradigmxyz/local_reth) repository, with some slight modifications
- expanded the documentation about running Reth with Docker

## Considerations

The default `docker-compose.yml` is meant to be a starting point and any reth configuration can be easily added by the end user through environment variables or command line args.

I'd love some feedback regarding a good default configuration for reth to include in the default `docker-compose.yml`.
i.e. in [local-reth](https://github.com/paradigmxyz/local_reth) the `RETH_TIP` env was set to sync only up to ETH block 5,000,000. I removed it since it seems like an opinionated choice that (imo) is not suited for a standard run configuration.

paolofacchinetti commented:
differs from PR #3490 by:
- keeping the monitoring stack from the `/etc` directory usable with a standalone, non Docker Reth installation.
- providing a standard docker-compose similar to the one in the local_eth repo meant as a starting point, with no Lighthouse container

onbjerg reviewed (approved):
thanks!

onbjerg reviewed (changes requested):
sorry, I was a bit too quick there; can you override the existing docker-compose we have in `etc` instead?

gakonst commented:
Could we also make this start Lighthouse alongside Reth?

paolofacchinetti commented:
@gakonst I should be able to do it today 

Just to clarify, do you want me to keep the old `docker-monitoring.yml` that would be useful to monitor non-containerized reth instances, or should I just override it with the new one?

paolofacchinetti commented:
@gakonst @onbjerg I moved the `docker-compose.yml` into `/etc`, overriding the existing `docker-monitoring.yml`. I also set the remote image as the default and updated docs.

Regarding Lighthouse and [ethereum-metrics-exporter](https://github.com/ethpandaops/ethereum-metrics-exporter), adding these would require enabling the http-rpc server on Reth and activating specific Rpc modules. That looks like an opinionated choice to me, wouldn't it be better to add it to the [local_reth](https://github.com/paradigmxyz/local_reth) repo instead? (also see https://github.com/paradigmxyz/reth/pull/3490#issuecomment-1616418577)

gakonst commented:
> Regarding Lighthouse and [ethereum-metrics-exporter](https://github.com/ethpandaops/ethereum-metrics-exporter), adding these would require enabling the http-rpc server on Reth and activating specific Rpc modules. That looks like an opinionated choice to me, wouldn't it be better to add it to the [local_reth](https://github.com/paradigmxyz/local_reth) repo instead? (also see https://github.com/paradigmxyz/reth/pull/3490#issuecomment-1616418577)

Yep it is a bit opinionated, ultimately fine with it if we pull it in a separate docker-compose.yml and one would need to do `docker-compose -f etc/el.yml -f etc/cl.yml up`? I would like the UX to be as close as possible to "clone reth repo, docker-compose up, forget about it"

gakonst commented on book/installation/docker.md:
````diff
@@ -45,3 +45,71 @@ The build will likely take several minutes. Once it's built, test it with:
+
+```bash
+docker run \
+    -v rethdata:/root/.local/share/reth/db \
````
love it

gakonst commented on etc/docker-compose.yml:
```diff
@@ -1,39 +1,63 @@
+  reth:
+    restart: unless-stopped
+    image: ghcr.io/paradigmxyz/reth
+    ports:
+      - '9000:9000'
```
what do we think about 8545? 30303? 8551 for CL?

paolofacchinetti replied:
depends on what we want the default to be. Maybe expose everything thus making it easier for people to just comment the ports they want to close? 

also fyi exposing some of these ports is useless unless the http rpc is active, so maybe we should enable it by default?

gakonst commented on etc/prometheus/prometheus.yml:
```diff
@@ -3,4 +3,4 @@ scrape_configs:
     metrics_path: "/"
     scrape_interval: 5s
     static_configs:
-      - targets: ['localhost:9001', 'host.docker.internal:9001']
```
think we made this 9001 bc CLs like LH use 9000

gakonst reviewed (approved):
supportive and we can iterate

onbjerg commented:
@gakonst @paolofacchinetti I added lighthouse and adjusted the ports - been a while since I worked with docker compose, so PTAL if something is off. Unfortunately reth does not currently "just" generate a JWT at the given path if a path is specified (and it is empty), so we have to generate the JWT before starting the stack

onbjerg reviewed (approved):
(marking as approved to unblock merge queue, awaiting a 2nd review)
